.github/workflows/tests.yml

```diff
@@ -474,6 +474,227 @@ jobs:
+          sudo LXD_DIR=/var/lib/lxd lxc network create local-network
+          sudo LXD_DIR=/var/lib/lxd lxc profile device add default eth0 nic network=local-network
+          sudo LXD_DIR=/var/lib/lxd lxc config set core.https_address "[::]:8443"
+          sudo LXD_DIR=/var/lib/lxd lxc config set cluster.https_address "127.0.0.1"
+          sudo LXD_DIR=/var/lib/lxd lxc cluster enable local
+          sudo LXD_DIR=/var/lib/lxd lxc config set user.show_permissions=true
+
+      - name: Checkout LXD-UI
+        uses: actions/checkout@v4
+        with:
+            repository: 'canonical/lxd-ui'
+            ref: main
+            path: lxd-ui
+
+      - name: Install Dotrun
+        run: sudo pip3 install dotrun
+
+      - name: Restore cached keys
+        uses: actions/cache/restore@v3
+        with:
+          path: lxd-ui/keys
+          key: keys-folder
+
+      - name: Install LXD-UI dependencies
+        run: |
+          set -eux
+          sudo chmod 0777 ./lxd-ui
+          cd lxd-ui
+          dotrun install
+
+      - name: Run LXD-UI
+        env:
+          ENVIRONMENT: devel
+          PORT: 8407
+          LXD_UI_BACKEND_IP: 172.17.0.1
+        run: |
+          set -eux
+          cd lxd-ui
+          dotrun &
+          curl --head --fail --retry-delay 2 --retry 100 --retry-connrefused --insecure https://localhost:8407
+
+      - name: Set keys permissions
+        run: |
+          set -eux
+          sudo chmod -R 0666 lxd-ui/keys
+          sudo chmod 0777 lxd-ui/keys
+
+      - name: Save keys
+        uses: actions/cache/save@v3
+        with:
+          path: lxd-ui/keys
+          key: keys-folder
+
+      - name: Allow LXD-UI keys
+        shell: bash
+        run: |
+          set -eux
+          sudo LXD_DIR=/var/lib/lxd lxc config trust add lxd-ui/keys/lxd-ui.crt
+
+      - name: Create a custom image
+        shell: bash
+        run: |
+          set -eux
+          sudo LXD_DIR=/var/lib/lxd lxc launch ubuntu-minimal:22.04 my-instance
+          sudo LXD_DIR=/var/lib/lxd lxc publish my-instance --alias my-custom-image --force
+          sudo LXD_DIR=/var/lib/lxd lxc delete my-instance --force
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Install Playwright Browsers
+        run: |
+          set -eux
+          cd lxd-ui
+          npx playwright install --with-deps chromium
+
+      - name: Create OIDC users
+        shell: bash
+        run: |
+          set -eux
+          sudo LXD_DIR=/var/lib/lxd PATH=/home/runner/go/bin:$PATH ./lxd-ui/tests/scripts/create_oidc_identities
+
+      - name: Run Playwright tests
+        run: |
+          set -eux
+          cd lxd-ui
+          CI=true DISABLE_VM_TESTS=true npx playwright test --project chromium:lxd-latest-edge
+          #npx playwright test --project chromium:lxd-5.21-edge
+          #npx playwright test --project lxd-5.0-edge
+
+      - name: Upload lxd-ui test artifacts
```

Comment:
lets set a short artifact lifetime on these please

Reply:
Good idea, set this to 14 days.